wdl-analysis/src/types/v1.rs

```diff
@@ -504,16 +504,20 @@ where
                     // Check for a sep option is specified; if so, accept `Array[P]` where `P` is
                     // primitive.
                     let mut coercible = false;
-                    if let Some(PlaceholderOption::Sep(_)) = placeholder.option() {
-                        if let Type::Compound(c) = ty {
-                            if let CompoundTypeDef::Array(a) =
-                                self.types.type_definition(c.definition())
-                            {
-                                if !a.element_type().is_optional()
-                                    && a.element_type().as_primitive().is_some()
+
+                    for option in placeholder.options() {
```

Comment:
@peterhuene this was the place that I think needed your attention the most.

Reply:
I think we can revert this along with the corresponding change to `Placeholder::options` in the AST.